src/EnergyPlus/WindowManager.cc

```diff
@@ -768,25 +790,103 @@ namespace WindowManager {
+						//  assuming wlt same as wle
+						int NumX2 = TableLookup( PerfCurve( Material( LayPtr ).GlassSpecAngTransDataPtr ).TableIndex ).NumX2Vars;
+						for ( ILam = 1; ILam <= NumX2; ++ILam ) {
+							wlt( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngTransDataPtr ).TableIndex ).X2( ILam );
+							t( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngTransDataPtr ).TableIndex ).Y( ILam );
+							rff( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngFRefleDataPtr ).TableIndex ).Y( ILam );
+							rbb( IGlass, ILam ) = TableData( PerfCurve( Material( LayPtr ).GlassSpecAngBRefleDataPtr ).TableIndex ).Y( ILam );
+						}						
+						SolarSprectrumAverage( t, tmpTrans );
+						SolarSprectrumAverage( rff, tmpReflectSolBeamFront );
+						SolarSprectrumAverage( rbb, tmpReflectSolBeamBack );
+
+						// visible properties
+						VisibleSprectrumAverage( t, tmpTransVis );
+						VisibleSprectrumAverage( rff, tmpReflectVisBeamFront );
+						VisibleSprectrumAverage( rbb, tmpReflectVisBeamBack );
+
+						// set this material to average spectral data
+						Material( LayPtr ).GlassSpectralAndAngle = false;
+						Material( LayPtr ).Trans = tmpTrans;
+						Material( LayPtr ).TransVis = tmpTransVis;
+						Material( LayPtr ).ReflectSolBeamFront = tmpReflectSolBeamFront;
+						Material( LayPtr ).ReflectSolBeamBack = tmpReflectSolBeamBack;
+						Material( LayPtr ).ReflectVisBeamFront = tmpReflectVisBeamFront;
+						Material( LayPtr ).ReflectVisBeamBack = tmpReflectVisBeamBack;
+						SpecDataNum = 0;
+					}
+				}
 			} // End of loop over glass layers in the construction for front calculation
 
+			// Perform do loop to find common incident angles if multiple SpectralAngular layers are specified
+			for ( IGlass = 1; IGlass <= NGlass; ++IGlass ) {
+				LayPtr = Construct( ConstrNum ).LayerPoint( LayerNum( IGlass ) );
+				if ( Material( LayPtr ).GlassSpectralAndAngle ) {
+					AngleNum( IGlass ) = LayerNum( IGlass );
+				}
+			}
+			SetCommonIncidentAngles( ConstrNum, NGlass, TotalIPhi, AngleNum );
+
+			for ( IGlass = 1; IGlass <= NGlass; ++IGlass ) {
+				LayPtr = Construct( ConstrNum ).LayerPoint( LayerNum( IGlass ) );
+				if ( Material( LayPtr ).GlassSpectralAndAngle ) {
+					CosPhiIndepVar = 0.0;
+					for ( IPhi = 1; IPhi <= TotalIPhi; ++IPhi ) {
+						Phi = TableLookup( PerfCurve( Material( LayPtr ).GlassSpecAngTransDataPtr ).TableIndex ).X1Var( IPhi );
+						CosPhiIndepVar( IPhi ) = std::cos( Phi * DegToRadians );
+					}
+					break;
+				}
+			}
+
 			// Loop over incidence angle from 0 to 90 deg in 10 deg increments.
 			// Get glass layer properties, then glazing system properties (which include the
 			// effect of inter-reflection among glass layers) at each incidence angle.
 
-			for ( IPhi = 1; IPhi <= 10; ++IPhi ) {
+			for ( IPhi = 1; IPhi <= TotalIPhi; ++IPhi ) {
 				Phi = double( IPhi - 1 ) * 10.0;
+				for ( IGlass = 1; IGlass <= NGlass; ++IGlass ) {
```

Comment:
A comment here would be good to explain that the normal value for Phi above gets overridden if there's a layer that is SpectralAndAngle.  And that TotalIPhi is 10 when it's not.